src/design-system/components/Box/Box.tsx

```diff
@@ -0,0 +1,179 @@
+import { flatten, mapValues } from 'lodash';
+import React, { forwardRef, ReactNode, useMemo } from 'react';
+import { View } from 'react-native';
+import { negativeSpace, NegativeSpace, space, Space } from '../../layout/space';
+import {
+  BackgroundProvider,
+  BackgroundProviderProps,
+} from '../BackgroundProvider/BackgroundProvider';
+import type * as Polymorphic from './polymorphic';
+
+const fraction = (numerator: number, denominator: number) =>
+  `${(numerator * 100) / denominator}%`;
+
+const widths = {
+  '1/2': fraction(1, 2),
```

Comment:
can't we make this just generic by parsing the text? Or caching some, but parsing the rest?

Reply:
I went with this approach partly because I wanted to make sure it was type checked, but we've upgraded TypeScript since I wrote this code and it's possible to use template literal types to cover this, e.g. type FractionString = `${number}/${number}`. I'll look into this.

Reply:
I might leave this one for now. In practice I feel like these are all the fractions we'll need and it keeps the code/types simpler while still being performant, but we can switch to something more complicated if it turns out we need more options in the future.